dev/benchmarks/macrobenchmarks/README.md

```diff
@@ -49,3 +51,43 @@ flutter drive --profile test_driver/post_backdrop_filter_perf.dart
 Results should be in the file `build/post_backdrop_filter_perf.timeline_summary.json`.
 
 More detailed logs should be in `build/post_backdrop_filter_perf.timeline.json`.
+
+## Web benchmarks
+
+Web benchmarks are compiled from the same entrypoint in `lib/web_benchmarks.dart`.
+
+### How to write a web benchmark
+
+Create a new file for your benchmark under `lib/src/web`. See `bench_draw_rect.dart`
+as an example.
+
+Choose one of the two benchmark types:
+
+* A "raw benchmark" records performance metrics from direct interactions with
+  `dart:ui` with no framework. This kind of benchmark is good for benchmarking
+  low-level engine primitives, such as layer, picture, and semantics performance.
+* A "widget benchmark" records performance metrics using a widget. This kind of
+  benchmark is good for measuring the performance of widgets, often together with
+  engine work that widget-under-test incurs.
+
+For a raw benchmark implement `RawRecorder` (tip: you can start by copying
```

Comment:
`s/implement/extend` ?

Reply:
Another thing that I'd love to see documented is where I should put "setup" code that prepares things for the benchmark but isn't included in the numbers.

Reply:
Done.